src/network/postgres/postgres_network_commands.cpp

```diff
@@ -210,7 +216,16 @@ Transition ParseCommand::Exec(const common::ManagedPointer<ProtocolInterpreter>
     out->WriteNoticeResponse("NOTICE:  we don't yet support that query type.");
   }
 
-  postgres_interpreter->SetStatement(statement_name, std::move(statement));
+  const auto fingerprint_result UNUSED_ATTRIBUTE = pg_query_fingerprint(query.c_str());
+  auto cached_statement = postgres_interpreter->LookupStatementInCache(fingerprint_result.hexdigest);
```

Comment:
Why is this lookup being done after parsing?

Reply:
My thinking was to make sure it parses properly before you try to fingerprint it, otherwise fingerprinting will fail and I'm not sure we want to rely on that as our parse validity first;